com.unity.render-pipelines.high-definition/Editor/RenderPipeline/HDShaderUtils.cs

```diff
@@ -83,13 +83,19 @@ internal enum ShaderID
         /// False: unknown shader used in material
         /// </returns>
         public static bool ResetMaterialKeywords(Material material)
+        {
+            var shaderID = GetShaderEnumFromShader(material.shader);
+            return ResetMaterialKeywords(material, shaderID);
+        }
+
+        internal static bool ResetMaterialKeywords(Material material, ShaderID shaderId)
```

Comment:
A new internal version of this function, so that VFX can manually provide the ShaderID from the current Shader Graph asset it is using. The original implementation of this function would fail to configure the material for VFX, since none of that meta data is stored in the VFX-generated shader.

Reply:
poking @slunity  for awareness of this coming change of ResetMaterialKeywords